addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/handler/AirQualityHandler.java

```diff
@@ -0,0 +1,321 @@
+        logger.debug("config refresh = {}", config.getRefresh());
+
+        boolean validConfig = true;
+
+        if (StringUtils.trimToNull(config.getApiKey()) == null) {
+            logger.error("Parameter 'apikey' is mandatory and must be configured, disabling thing '{}'",
+                    getThing().getUID());
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.getLocation()) == null && config.getStationId() == null) {
+            logger.error(
+                    "Parameter 'location' or 'stationId' is mandatory and must be configured, disabling thing '{}'",
+                    getThing().getUID());
+            validConfig = false;
+        }
+        if (config.getRefresh() != null && config.getRefresh() < 5) {
+            logger.error("Parameter 'refresh' must be at least 5 minutes, disabling thing '{}'", getThing().getUID());
+            validConfig = false;
+        }
+
+        if (validConfig) {
+            startAutomaticRefresh();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    /**
+     * Start the job refreshing the Air Quality data
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request new air quality data to the aqicn.org service
+                        aqiResponse = updateAirQualityData();
+
+                        // Update all channels from the updated AQI data
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId(), aqiResponse);
+                        }
+                    } catch (Exception e) {
+                        logger.error("Exception occurred during execution: {}", e.getMessage());
+                    }
+                }
+            };
+
+            AirQualityConfiguration config = getConfigAs(AirQualityConfiguration.class);
+            int period = (config.getRefresh() != null) ? config.getRefresh().intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleAtFixedRate(runnable, 0, period, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Air Quality handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId(), aqiResponse);
+        } else {
+            logger.warn("The Air Quality binding is a read-only and can not handle command {}", command);
```

Comment:
Imho, this should only be debug. Maybe the channel is linked to an item, which is also linked to other bindings, which accept commands. So sending a command to the item might be fully correct - your binding should simply ignore it.